internal/surface/backend/adapter.go

```
@@ -0,0 +1,333 @@
+		return fmt.Errorf("%w: %w", ErrInvalidStartSpec, ErrInvalidRecoveryTag)
+	}
+	if !s.bootstrap.Set() {
+		return fmt.Errorf("%w: no bootstrap command", ErrInvalidStartSpec)
+	}
+	return nil
+}
+
+// safeText bounds a string that will reach a manager's command line and UI:
+// length-capped, valid UTF-8, and free of the control sequences that would let
+// it rewrite a terminal it was only supposed to appear in.
+func safeText(s string, max int, what string) error {
```

golangci-lint (2.12.2)

[error] param max has same name as predeclared identifier

(predeclared)